tests/bdd/test_volume_nexus.py

```diff
@@ -0,0 +1,167 @@
+    # Brings back the mayastor instance, which should expose the replicas upon pool import
+    Docker.restart_container(MAYASTOR_2)
+    check_replicas_online()
+
+
+@then("it shall be recreated")
+def it_shall_be_recreated():
+    """it shall be recreated."""
+    check_nexus_online()
+
+
+@then("the nexus target shall be removed from its associated node")
+def the_nexus_target_shall_be_removed_from_its_associated_node():
+    """the nexus shall be removed from its associated node."""
+    check_nexus_removed()
+
+
+""" ... """
+
+VOLUME_UUID = "5cd5378e-3f05-47f1-a830-a0f5873a1449"
+VOLUME_SIZE = 10485761
+NUM_VOLUME_REPLICAS = 1
+
+MAYASTOR_1 = "mayastor-1"
+MAYASTOR_2 = "mayastor-2"
+
+POOL_DISK1 = "cdisk1.img"
+POOL1_UUID = "4cc6ee64-7232-497d-a26f-38284a444980"
+POOL_DISK2 = "cdisk2.img"
+POOL2_UUID = "4cc6ee64-7232-497d-a26f-38284a444990"
+
+
+@pytest.fixture(scope="function")
+def create_pool_disk_images():
+    # When starting Mayastor instances with the deployer a bind mount is created from /tmp to
+    # /host/tmp, so create disk images in /tmp
+    for disk in [POOL_DISK1, POOL_DISK2]:
+        path = f"/tmp/{disk}"
+        with open(path, "w") as file:
+            file.truncate(20 * 1024 * 1024)
+
+    yield
+    for disk in [POOL_DISK1, POOL_DISK2]:
+        path = f"/tmp/{disk}"
+        if os.path.exists(path):
+            os.remove(path)
+
+
+@pytest.fixture(scope="function")
+def init(create_pool_disk_images):
+    # Shorten the reconcile periods and cache period to speed up the tests.
+    Deployer.start_with_args(
+        [
+            "-j",
+            "-m=2",
+            "-w=10s",
+            "--reconcile-idle-period=500ms",
+            "--reconcile-period=500ms",
+            "--cache-period=1s",
+        ]
+    )
+
+    # Create pools
+    ApiClient.pools_api().put_node_pool(
+        MAYASTOR_2,
+        POOL2_UUID,
+        CreatePoolBody([f"aio:///host/tmp/{POOL_DISK2}"], labels={"node": MAYASTOR_2}),
+    )
+
+    yield
+    Deployer.stop()
+
+
+@retry(wait_fixed=200, stop_max_delay=5000)
+def check_target_faulted():
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert NexusState(volume.state.target["state"]) == NexusState("Faulted")
+
+
+@retry(wait_fixed=200, stop_max_attempt_number=30)
+def check_nexus_online():
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    nexus_uuid = volume.state.target["uuid"]
+    nexus = ApiClient.nexuses_api().get_nexus(nexus_uuid)
+    assert nexus.state == NexusState("Online")
+
+
+@retry(wait_fixed=200, stop_max_delay=5000)
+def check_nexus_removed():
+    volume = ApiClient.volumes_api().get_volume(VOLUME_UUID)
+    assert (
+        not hasattr(volume.state, "target")
+        # or it might have been recreated if we lost the "race"...
+        or NexusState(volume.state.target["state"]) == NexusState("Online")
+        or NexusState(volume.state.target["state"]) == NexusState("Degraded")
```

Comment:
I understand why this is necessary, but I wonder if we could make this check more robust somehow. I know it's unlikely, but what happens if, in the future, there was the possibility that the nexus got set from "Faulted" to another state without being recreated.